libcore/src/IO/EventFileParser.cpp

```diff
@@ -0,0 +1,317 @@
+/**
+ * Copyright (c) 2020 Forschungszentrum Jülich GmbH. All rights reserved.
+ *
+ * \section License
+ * This file is part of JuPedSim.
+ *
+ * JuPedSim is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * any later version.
+ *
+ * JuPedSim is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with JuPedSim. If not, see <http://www.gnu.org/licenses/>.
+ **/
+#include "EventFileParser.h"
+
+#include "general/Logger.h"
+#include "general/Macros.h"
+#include "tinyxml.h"
+
+std::vector<Event> EventFileParser::ParseEvents(const fs::path & eventFile)
+{
+    LOG_INFO("Parsing event file");
+    TiXmlDocument docEvent(eventFile.string());
+    if(!docEvent.LoadFile()) {
+        LOG_ERROR("Cannot parse event file: {}", docEvent.ErrorDesc());
+        return std::vector<Event>();
+    }
+
+    TiXmlElement * xRootNode = docEvent.RootElement();
+    if(!xRootNode) {
+        LOG_ERROR("Event file root element missing");
+        return std::vector<Event>();
+    }
+
+    if(xRootNode->ValueStr() != "JPScore") {
+        LOG_ERROR("Event file root element should be 'JPScore'");
+        return std::vector<Event>();
+    }
+
+    TiXmlNode * xEvents = xRootNode->FirstChild("events");
+    if(!xEvents) {
+        LOG_ERROR("No events found");
+        return std::vector<Event>();
+    }
+
+    std::vector<Event> events;
+    for(TiXmlElement * e = xEvents->FirstChildElement("event"); e;
+        e                = e->NextSiblingElement("event")) {
+        // Read id and check for correct value
+        int id;
+        if(const char * attribute = e->Attribute("id"); attribute) {
+            if(int value = xmltoi(attribute, -1);
+               value > -1 && attribute == std::to_string(value)) {
+                id = value;
+            } else {
+                LOG_ERROR("event: id is set but no integer");
+                continue;
+            }
+        } else {
+            LOG_ERROR("event: id required");
+            continue;
+        }
+
+        // Read time of events
+        double time;
+        if(const char * attribute = e->Attribute("time"); attribute) {
+            if(double value = xmltof(attribute, std::numeric_limits<double>::min()); value >= 0.) {
+                time = value;
+            } else {
+                LOG_ERROR("event {:d}: time not set properly", id);
+                continue;
+            }
+        } else {
+            LOG_ERROR("event {:d}: time required", id);
```

Comment:
`{:d}` is the default for formatting integers. I am not sure if we should keep the format specifier or use empty braces